plugins/inputs/mikrotik/parsers.go

```diff
@@ -0,0 +1,103 @@
+package mikrotik
+
+import (
+	"fmt"
+	"regexp"
+	"slices"
+	"strings"
+	"time"
+)
+
+func parse(data common) (points []parsedPoint, err error) {
+	var errorList []string
+
+	data = filterCommon(data)
+
+	for i := range data {
```

Comment:
Why not filter the data in the loop below? This would save one iteration over all data.